Add pytests for FANS (#55)
DataAnalyticsEngineering/FANS: test/README.md

# Tests

> This directory contains tests for the FANS. Tests serve two purposes:

Reviewer: ```suggestion
This directory contains tests for FANS. Tests are primarily of two types
```

1. Verify the correct functioning of the code.
2. Provide example problems that demonstrate how FANS can be used.

## Test Structure

The test directory includes:

- `CMakeLists.txt` - Configures tests for CTest integration
- `input_files/` - Example input JSON files for each test case
- Python based validation tests in `pytest/` directory for validating results

## Running Tests

### Using CTest

> Tests are configured by the main CMake system. To run all FANS tests:

Reviewer: ```suggestion
Tests are configured by CMake when FANS is built. If your CMake build folder is named `build`, run the FANS tests in the following way
```

```bash
cd ../build
ctest fans
```

This will run all test cases and generate result files in `build/test/`.

## Result Validation

> After running the tests, verify the results using pytests in the `pytest/` directory with the following command:

Reviewer: ```suggestion
After running the tests, the results are verified using pytest. Go to the directory `test/pytest/` and run `pytest`. We recommend running pytest via a pre-configured pixi setup
```

```bash
pixi run test
```

> Note: The pytest validation expects result files to be in `build/test/` directory, so make sure to run the tests first.

Reviewer: ```suggestion
Note: The validation tests expect result files to be in `build/test/` directory, so make sure to run the tests first.
```
